src/main/groovy/com/bmuschko/gradle/java2html/ConvertCodeTask.groovy

```diff
@@ -15,38 +15,48 @@
  */
 package com.bmuschko.gradle.java2html
 
-
 import org.gradle.api.DefaultTask
 import org.gradle.api.file.FileCollection
+import org.gradle.api.tasks.CacheableTask
+import org.gradle.api.tasks.Classpath
 import org.gradle.api.tasks.Input
 import org.gradle.api.tasks.InputFiles
+import org.gradle.api.tasks.Optional
 import org.gradle.api.tasks.OutputDirectory
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
 import org.gradle.api.tasks.TaskAction
 
 /**
  * Java2HTML files documentation generation task.
  *
  * @author Benjamin Muschko
  */
+@CacheableTask
 class ConvertCodeTask extends DefaultTask {
-    @InputFiles FileCollection classpath
+    @Classpath FileCollection classpath
+
+    @PathSensitive(PathSensitivity.RELATIVE)
     @InputFiles FileCollection srcDirs
+
+    @PathSensitive(PathSensitivity.RELATIVE)
     @OutputDirectory File destDir
+
     @Input String includes
-    @Input String outputFormat
-    @Input Integer tabs
-    @Input String style
-    @Input Boolean showLineNumbers
-    @Input Boolean showFileName
-    @Input Boolean showDefaultTitle
-    @Input Boolean showTableBorder
-    @Input Boolean includeDocumentHeader
-    @Input Boolean includeDocumentFooter
-    @Input Boolean addLineAnchors
-    @Input String lineAnchorPrefix
-    @Input String horizontalAlignment
-    @Input Boolean useShortFileName
-    @Input Boolean overwrite
+    @Input @Optional String outputFormat
```

Comment:
I had a look at the plugin code again and the task. At the moment we don't have any handling in place that would _not_ pass the attributes to Ant if the value is null. Not sure how the Ant task would behave. Let's remove `@Optional` to ensure that a value is being enforced.

Reply:
OK